hack/run-locally.sh

```diff
@@ -31,13 +31,41 @@ function override_install_manifests() {
     fi
 }
 
-# Extract environment variables for the CNO DaemonSet
+# Extract and update CNO environment variables
 function setup_operator_env() {
-    if [[ manifests/0000_70_cluster-network-operator_03_daemonset.yaml -nt "${CLUSTER_DIR}/env.sh" ]]; then
-        echo "Copying environment variables from manifest to ${CLUSTER_DIR}/env.sh"
-        oc --kubeconfig=hack/null-kubeconfig patch --local=true -f manifests/0000_70_cluster-network-operator_03_daemonset.yaml -p '{}' -ojsonpath='{range .spec.template.spec.containers[0].env[?(@.value)]}{.name}{"="}{.value}{"\n"}' > "${CLUSTER_DIR}/env.sh"
+    local mode="$1"
+    local image_env_key="$2"
+    local plugin_image="$3"
+
+    if [[ "${mode}" == "attach" ]]; then
+        # Extract environment from a running cluster
+        if [[ ! -e "${CLUSTER_DIR}/env.sh" ]]; then
+            echo "Copying environment variables from manifest to ${CLUSTER_DIR}/env.sh"
+            kubectl get deployment -n openshift-network-operator network-operator -ojsonpath='{range .spec.template.spec.containers[0].env[?(@.value)]}{.name}{"="}{.value}{"\n"}' > "${CLUSTER_DIR}/env.sh"
+        fi
+    else
+        # Extract environment from the CNO's manifests
+        if [[ manifests/0000_70_cluster-network-operator_03_daemonset.yaml -nt "${CLUSTER_DIR}/env.sh" ]]; then
+            echo "Copying environment variables from manifest to ${CLUSTER_DIR}/env.sh"
+            oc --kubeconfig=hack/null-kubeconfig patch --local=true -f manifests/0000_70_cluster-network-operator_03_daemonset.yaml -p '{}' -ojsonpath='{range .spec.template.spec.containers[0].env[?(@.value)]}{.name}{"="}{.value}{"\n"}' > "${CLUSTER_DIR}/env.sh"
+        fi
     fi
+
     sed -i -e "s/^RELEASE_VERSION=.*/RELEASE_VERSION=${RELEASE_VERSION}/" "${CLUSTER_DIR}/env.sh"
+
+    if [[ -n "${plugin_image}" ]]; then
+        sed -i -e "s#^${image_env_key}=.*#${image_env_key}=${plugin_image}#" "${CLUSTER_DIR}/env.sh"
+    fi
+}
+
+# Patch the CNO out of the cluster-version-operator and scale it down to zero so
+# we can run our local CNO instead
+function stop_deployed_operator() {
+    echo "Telling the CVO to ignore the network operator"
+    kubectl patch --type=json -p "$(cat hack/overrides-patch.yaml)" clusterversion version
```

Comment:
So, tying into my earlier comment about running the script, killing it, rebuilding, and running again, note that the existing code goes out of its way to not print messages claiming it's doing something if that thing has already been done. (ie, ideally, if you attach to a cluster that is already not running a CNO, it shouldn't tell you that it's stopping the CNO.)